guides/plugins/plugins/checkout/payment/add-payment-plugin.md

```diff
@@ -527,14 +733,79 @@ class SwagBasicExample extends Plugin
 }
 ```
 
-In the `install` method, you start by creating a new payment method, if it doesn't exist yet. If you need to know what's happening in there, you might want to have a look at our guide regarding [Writing data](../../framework/data-handling/writing-data).
+:::
 
-::: warning
-**Do not** do the opposite in the `uninstall` method and remove the payment method. This might lead to data inconsistency if the payment method was used in some orders. Instead, only deactivate the method!
+In the `install` method, you start by creating a new payment method, if it doesn't exist yet.
+If you need to know what's happening in there, you might want to have a look at our guide regarding [Writing data](../../framework/data-handling/writing-data).
+
+::: danger
+**Do not** do the opposite in the `uninstall` method and remove the payment method. 
+This might lead to data inconsistency if the payment method was used in some orders. 
+Instead, only deactivate the method!
 :::
 
 The `activate` method and `deactivate` method just do that, activating and deactivating the payment method, respectively.
 
 ### Identify your payment
 
-You can identify your payment by the entity property `formattedHandlerIdentifier`. It shortens the original handler identifier \(php class reference\): `Custom/Payment/SEPAPayment` to `handler_custom_sepapayment`. The syntax for the shortening can be looked up in [Shopware\Core\Checkout\Payment\DataAbstractionLayer\PaymentHandlerIdentifierSubscriber](https://github.com/shopware/shopware/blob/v6.3.4.1/src/Core/Checkout/Payment/DataAbstractionLayer/PaymentHandlerIdentifierSubscriber.php).
+You can identify your payment by the entity property `formattedHandlerIdentifier`. 
+It shortens the original handler identifier \(php class reference\): `Custom/Payment/SEPAPayment` to `handler_custom_sepapayment`. 
+The syntax for the shortening can be looked up in [Shopware\Core\Checkout\Payment\DataAbstractionLayer\PaymentHandlerIdentifierSubscriber](https://github.com/shopware/shopware/blob/v6.3.4.1/src/Core/Checkout/Payment/DataAbstractionLayer/PaymentHandlerIdentifierSubscriber.php).
+
+Otherwise, you can use your given technical name to uniquely identify your payment method.
+
+## Migrating payment handlers from 6.6
+
+If you are migrating a payment handler from a version before 6.7, you have to adjust you payment handlers.
```

Comment:
This sentences is no help to anyone. Either add something like `You need to move from the existing interfaces to the new abstract class and add your own order data loading.` or remove it.